### ISSUE TITLE
Provide config option for short cache paths

### ISSUE THREAD

**Issue by ambo** [berlinonline/converjon#138]
There are issues with image paths beeing too long under certain conditions. To avoid those paths I suggest to switch to a md5 hash to create the local file path instead of base64. 

To ensure backwards compatibility, we should provide a config option which uses the current method by default, but lets you turn on the short paths.

**Comment by graste** [comment]
We could try this:

```diff
diff --git a/lib/pathutils.js b/lib/pathutils.js
index 54eeddf..27972c7 100644
--- a/lib/pathutils.js
+++ b/lib/pathutils.js
@@ -39,7 +39,8 @@ exports.getHashPath = function(input) {
         hash = hash.substr(8);
     } while (part !== "");
 
-    return join(path_parts);
+    // join array parts into string, split into maximum N char chunks, re-join as string
+    return join(join(path_parts).match(/.{1,20}/g));
 };
 
 exports.getOptionsPath = function(options) {
```
The "20" characters per path part is a guess to get 2.5x less path depth!? Don't have docker/npm installed to test it atm. The probably failing test should probably be fixed as well. :-) @trendfischer @neffets

**Comment by trendfischer** [comment]
@graste I would use a MD5, so we can be sure, the path would never be longer than 32 chars + path separator.

```js
var crypto = require('crypto');
return crypto.createHash('md5').update(join(path_parts).digest("hex");
```

Well I believe this piece of code can be optimized, it's just to show what I mean.

**Comment by graste** [comment]
Depends on how many files per folder or how many levels of folders we'd like to have. I'd prefer to have more than 1 level.

Boils down to: decide, implement, test, commit, push and we'll create a release. Would probably be 2.7.9 as it's a bugfix imho.

**Comment by graste** [comment]
Decided to implement an example in a branch that we might release as 2.7.9 if @trendfischer @neffets etc.  like the changes. See https://github.com/berlinonline/converjon/tree/issue-138

New `cache` folder structure is:

```txt
$ tree cache/
cache/
├── 3d38cda2f60d38c45b719c71128fc7d438a19e828d7cf4ac5f8c65d52a8be95c
│   ├── meta
│   ├── source
│   └── targets
│       ├── b24a174707f0eb7c6fa6ade7db303a767dfc6370af277f9f351c2cc0c0282b7a
│       └── __no_options
└── c0ee1e228a5e35561d39774005aa581eabe78903a810e84ecb4c56892c378ba3
    ├── meta
    ├── source
    └── targets
        ├── b24a174707f0eb7c6fa6ade7db303a767dfc6370af277f9f351c2cc0c0282b7a
        ├── dc95f52f1b3c5b20e20dab0afc79d981a9aaf7141f504939b76fafb2ed442852
        └── fd75cb709ec1c0b4ab756dd231ba945a950032b63bbb9a88c2fab32e2ee606ad

4 directories, 9 files
```

This means: one folder per source url and 1 file in "targets" folder in that folder per requested conversion format (depending on url options, source file staleness and cache and garbage collection settings).

**Comment by graste** [comment]
After consideration w/ other we use a maximum of 256 folders in the 1st hierarchy level of the cache directory. This should ease cases where someone accidentally tries to cleanup the cache via shell expansion like ``rm cache/*``…
I'll try to merge the changes into master and release this as version 2.7.9.

**Comment by graste** [comment]
Released as 2.7.9.
